json reference field with omitempty can't compile

### Describe the bug

Code: https://play.vlang.io/p/c7eaaa5ba1

```v
import json

struct Number {
	min int
	max int
}

pub struct Resp {
pub:
	options []string                @[omitempty]
	number  &Number = unsafe{ nil } @[omitempty]
}

fn (r Resp) str() string {
	return json.encode(r)
}

fn main() {
	r1 := Resp{ options:['first','second']}
	r2 := Resp{ number:&Number{0,0} }
	r3 := Resp{ number:&Number{1,2} }
	assert r1.str() == '{"options":["first","second"]}'
	assert r2.str() == '{"number":{"min":0,"max":0}}'
	assert r3.str() == '{"number":{"min":1,"max":2}}'
}
```

### Reproduction Steps

Code mentioned above and in playground fails to compile. By commenting the reference field `omitempty` attribute the program works:

```v
	number  &Number = unsafe{ nil } // @[omitempty]
```

Seems any nil reference is already considered for omission, but anyhow adding the attribute the compilation fails.

### Expected Behavior

Compile and work with attribute

### Current Behavior

Output:

```
/tmp/v_60000/code.01JBHVSG4JPE527ZRNK1J1M331.tmp.c:2364: error: cannot convert 'struct main__Number *' to 'struct main__Number'
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
Exited with error status 1
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.8 a189d7f.d48cb18

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.8 a189d7f.d48cb18
OS: linux, Debian GNU/Linux 12 (bookworm) (VM)
Processor: 2 cpus, 64bit, little endian, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz

getwd: /home/admin/playground
vexe: /home/admin/v/v
vexe mtime: 2024-10-31 17:25:29

vroot: OK, value: /home/admin/v
VMODULES: OK, value: .vmodules
VTMP: OK, value: /tmp/v_0

Git version: git version 2.39.5
Git vroot status: Error: fatal: detected dubious ownership in repository at '/home/admin/v'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v
.git/config present: true

CC version: cc (Debian 12.2.0-14) 12.2.0
thirdparty/tcc status: Error: fatal: detected dubious ownership in repository at '/home/admin/v/thirdparty/tcc'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v/thirdparty/tcc
 Error: fatal: detected dubious ownership in repository at '/home/admin/v/thirdparty/tcc'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v/thirdparty/tcc
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzIzYzg4YTI3Y2M3NjgxYzYyNTc5ZDkiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.egEMnFX0IsuUAAwPm1mw7_6v2PCPMi4oPNR4GyxSlvk">Huly&reg;: <b>V_0.6-21164</b></a></sub>

--- a/vlib/json/json_encode_ptr_test.v
+++ b/vlib/json/json_encode_ptr_test.v
@@ -1,0 +1,31 @@
+import json
+
+struct Number {
+	min int
+	max int
+}
+
+pub struct Resp {
+pub:
+	options []string @[omitempty]
+	number  &Number = unsafe { nil }  @[omitempty]
+}
+
+fn (r Resp) str() string {
+	return json.encode(r)
+}
+
+fn test_main() {
+	r1 := Resp{
+		options: ['first', 'second']
+	}
+	r2 := Resp{
+		number: &Number{0, 0}
+	}
+	r3 := Resp{
+		number: &Number{1, 2}
+	}
+	assert r1.str() == '{"options":["first","second"]}'
+	assert r2.str() == '{"number":{"min":0,"max":0}}'
+	assert r3.str() == '{"number":{"min":1,"max":2}}'
+}

--- a/vlib/v/gen/c/json.v
+++ b/vlib/v/gen/c/json.v
@@ -842,7 +842,8 @@ fn (mut g Gen) gen_struct_enc_dec(utyp ast.Type, type_info ast.TypeInfo, styp st
 			} else if field.typ == ast.string_type {
 				enc.writeln('${indent}if (${prefix_enc}${op}${c_name(field.name)}.len != 0)')
 			} else {
-				if field_sym.kind in [.alias, .sum_type, .map, .array, .struct] {
+				if !field.typ.is_ptr()
+					&& field_sym.kind in [.alias, .sum_type, .map, .array, .struct] {
 					ptr_typ := g.equality_fn(field.typ)
 					if field_sym.kind == .alias {
 						enc.writeln('${indent}if (!${ptr_typ}_alias_eq(${prefix_enc}${op}${c_name(field.name)}, ${g.type_default(field.typ)}))')

